python/ray/_private/node.py

```diff
@@ -329,9 +329,15 @@ def __init__(
                 self.gcs_address,
                 self._raylet_ip_address,
             )
+            self._node_id = node_info["node_id"]
             if self._ray_params.node_manager_port == 0:
                 self._ray_params.node_manager_port = node_info["node_manager_port"]
 
+        # If it is a head node, try validating if
+        # external storage is configurable.
+        if head:
+            self.validate_external_storage()
+        self.destroy_external_storage()
```

Comment:
we should only do it if it's not `connect_only`

Reply:
Good catch.